packages/mui-material/src/TextField/TextField.d.ts

```diff
@@ -270,4 +276,12 @@ export type TextFieldProps = StandardTextFieldProps | FilledTextFieldProps | Out
  * - [TextField API](https://mui.com/material-ui/api/text-field/)
  * - inherits [FormControl API](https://mui.com/material-ui/api/form-control/)
  */
-export default function TextField(props: TextFieldProps): JSX.Element;
+export default function TextField<V extends Variants>(
+  props: {
+    /**
+     * The variant to use.
+     * @default 'outlined'
+     */
+    variant?: V;
+  } & Omit<TextFieldProps, 'variant'>,
+): JSX.Element;
```

Comment:
Why you need to do this for intellisense to work?

Reply:
without adding prop description for varaint here, `yarn docs:api` is not able to extract `variant` description. so i added here.

BTW, this is not required for making intellinse to work, changing `TextFieldProps  type` from union type to conditional type worked.

uninon type in master: https://github.com/mui/material-ui/blob/master/packages/mui-material/src/TextField/TextField.d.ts#L229

conditional type in this branch: https://github.com/mui/material-ui/blob/0d51f44779233d05b1519e69f2bbaac1fbd43e09/packages/mui-material/src/TextField/TextField.d.ts#L231

Reply:
> BTW, this is not required for making intellinse to work, changing TextFieldProps type from union type to conditional type worked.

Are you sure? Because I kept conditional type above and changed this back to:
`export default function TextField(props: TextFieldProps): JSX.Element;`

Intellisense does not work.

> without adding prop description for varaint here, yarn docs:api is not able to extract variant description. so i added here.

I kept it as `export default function TextField(props: TextFieldProps): JSX.Element`, there is no change in variant docs after doing `docs:api`.

Reply:
okay i see the confusion here, i didn't explained it correctly. By conditional type, i mean typescript should know the value of `valriant` passed in props to extract correct `TextFieldProps` type.

when `TextField` is defined like below code, we are passing `variant` type to conditional type, so that typescript is able to figure out which props to apply for `TextFieldProps` 
```

export default function TextField<V extends Variants>(
  props: {
    /**
     * The variant to use.
     * @default 'outlined'
     */
    variant?: V;
  } & Omit<TextFieldProps, 'variant'>,
): JSX.Element;
```

But if you define `TextField` like below code, typescript can't figure out what is value of `variant` passed and hence it can't extract correct type for `TextFieldProps` and hence intellisense doesn't work properly.

export default function TextField(props: TextFieldProps): JSX.Element;